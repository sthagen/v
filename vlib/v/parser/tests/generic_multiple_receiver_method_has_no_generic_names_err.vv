struct Queue<A, B>{
	buffer []A
	size   B
}

fn (q Queue<A, B>) poll() A {
	return q.buffer[0]
}

fn main() {
	q := Queue<string, int>{}
	println(q)
}
