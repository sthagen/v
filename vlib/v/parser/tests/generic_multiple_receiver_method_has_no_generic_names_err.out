vlib/v/parser/tests/generic_multiple_receiver_method_has_no_generic_names_err.vv:6:20: error: generic receiver method `poll` should add generic names, e.g. poll<T>
    4 | }
    5 |
    6 | fn (q Queue<A, B>) poll() A {
      |                    ~~~~
    7 |     return q.buffer[0]
    8 | }
